src/compas/geometry/primitives/polygon.py

```diff
@@ -156,12 +156,15 @@ def __getitem__(self, key):
         return self.points[key]
 
     def __setitem__(self, key, value):
-        self.points[key] = value
+        self.points[key] = Point(*value)
+        self._lines = None
 
     def __iter__(self):
         return iter(self.points)
 
     def __eq__(self, other):
+        if not hasattr(other, '__iter__') or not hasattr(other, '__len__') or len(self) != len(other):
+            return False
         return all(a == b for a, b in zip(self, other))
```

Comment:
we are not being consistent with the precision checks. perhaps we could use `allclose` everywhere?

Reply:
Can change this one, too.